deap/tools/crossover.py

```diff
@@ -428,17 +443,62 @@ def cxESTwoPoint(ind1, ind2):
 
     return ind1, ind2
 
+
 def cxESTwoPoints(ind1, ind2):
     """
     .. deprecated:: 1.0
         The function has been renamed. Use :func:`cxESTwoPoint` instead.
     """
     return cxESTwoPoint(ind1, ind2)
 
+
+######################################
+# ES Crossovers                      #
+######################################
+
+def semantic_crossover(ind1, ind2, gen_func=gp.genGrow, pset=None, min=2, max=6):
+    """
+    Implementation of the Semantic Crossover operator
+    :param ind1: first parent
+    :param ind2: second parent
+    :param gen_func: function responsible for the generation of the random tree that will be used during the mutation
+    :param pset: Primitive Set, which contains terminal and operands to be used during the evolution
+    :param min: min depth of the random tree
+    :param max: max depth of the random tree
+    :return: offsprings
+    """
+    for p in ['lf', 'mul', 'add', 'sub']:
```

Comment:
This seems a little restrictive, I havent read the paper but can it be generalized to other operators?

Reply:
Unfortunately, no. The point is that, using these operators, the fitness landscape is unimodal